lib/install/action/extract.js

```diff
@@ -68,14 +68,11 @@ function stageBundledModule (bundler, child, staging, parentPath, next) {
   var stageFrom = path.join(parentPath, 'node_modules', child.package.name)
   var stageTo = moduleStagingPath(staging, child)
 
-  // if this module bundles its own deps and it's not coming from OUR bundle then
-  // we're not responsible for staging its children.
-  if (!child.fromBundle && child.package.bundleDependencies) return finishModule()
```

Comment:
Skipping walking the kids like this would have resulted in breaking modules that had ina rather complicated set of interleaved bundled dependencies.

If one module with bundles depended on another with bundles and we ended up deciding to replace the second module, but not all of its children, then some of the children would come from the first module's bundled deps and some would come from the second.